### DISCUSSION TITLE
Item struct variable name revision

### DISCUSSION THREAD

**Discussion by kphoenix137** [diasurgical/DevilutionX#7303] [General]
Suggested names to rename these variables to. All variable names are kept between 3-20 characters. I opted for full words when possible for clarity, and I'm assuming most modern IDEs have an autofill/autocomplete feature that makes it less tedious to type longer variable names.

`_iSeed` -> `seed`
`_iCreateInfo` -> `creationFlags`
`_itype` -> `type`
`_iAnimFlag` -> `isAnimated` (Note: this is a boolean, so variable name should indicate as such for consistency in this codebase)
`position`
`AnimInfo` -> `animationInfo`
`_iDelFlag` -> REMOVED (Note: unused)
`_iPostDraw` -> `postDraw`
`_iIdentified` -> `isIdentified`
`_iMagical` -> `isMagical`
`_iName` -> `name`
`_iIName` -> `identifiedName`
`_iLoc` -> `equipLocation`
`_iClass` -> `class`
`_iCurs` -> `cursorGraphic`
`_ivalue` -> `value`
`_iIvalue` -> `idenfitiedValue`
`_iMinDam` -> `minDamage`
`_iMaxDam` -> `maxDamage`
`_iAC` -> `armorClass`
`_iFlags` -> `flags`
`_iMiscId` -> `miscID`
`_iSpell` -> `spell`
`IDidx` -> `tableID`
`_iCharges` -> `charges`
`_iMaxCharges` -> `maxCharges`
`_iDurability` -> `durability`
`_iMaxDur` -> `maxDurability`
`_iPLDam` -> `damagePercent`
`_iPLToHit` -> `toHit`
`_iPLAC` -> `armorClass`
`_iPLStr` -> `strength`
`_iPLMag` -> `magic`
`_iPLDex` -> `dexterity`
`_iPLVit` -> `vitality`
`_iPLFR` -> `resistFire`
`_iPLLR` -> `resistLightning`
`_iPLMR` -> `resistMagic`
`_iPLMana` -> `mana`
`_iPLHP` -> `life`
`_iPLDamMod` -> `damage`
`_iPLGetHit` -> `damageFromEnemies`
`_iPLLight` -> `lightRadius`
`_iSplLvlAdd` -> `spellLevel`
`_iRequest` -> `isBeingPickedUp`
`_iUid` -> `uniqueID`
`_iFMinDam` -> `minFireDamage`
`_iFMaxDam` -> `maxFireDamage`
`_iLMinDam` -> `minLightningDamage`
`_iLMaxDam` -> `maxLightningDamage`
`_iPLEnAc` -> `armorPierce`
`_iPrePower` -> `prefix`
`_iSufPower` -> `suffix`
`_iVAdd1` -> `addedValue1`
`_iVMult1` -> `valueMultiplier1`
`_iVAdd2` -> `addedValue2`
`_iVMult2` -> `valueMultiplier2`
`_iMinStr` -> `requiredStrength`
`_iMinMag` -> `requiredMagic`
`_iMinDex` -> `requiredDexterity`
`_iStatFlag` -> `canBeEquipped`
`_iDamAcFlags` -> `hellfireItemFlags`
`dwBuff` -> `extraFlags`

**Comment by ephphatha** [comment]
`hasAnimationFlag` > `isAnimated`, this is used for items that have just been dropped (and are playing the flip animation) or the magic rock, which has a repeating animation.
`request` > `isBeingPickedUp`, this is a flag set when a player claims an item and attempts to pick it up

`selectionFlag` > I don't have a good name for this but this is a combination of two properties about how players can interact with an object. The values are `0` = cannot be interacted with (usually dropped items in the middle of the drop animation), `1` = can be interacted with at ground level, `2` = can be interacted with at elevation, `3` = `1 | 2` = can be interacted with at both ground level and elevation (doesn't appear to actually be used).

Value `2` is used for the magic rock while it's using the second half of the sprite sheet, which renders the item at a height that makes it look like it's sitting on the pedestal. As this doesn't use the ground level flag the player can't pick the item up by hovering over the tile the object is actually sitting on, instead the cursor highlight logic tests the tile below the cursor making it so the player needs to hover over the tile above the rock/pedestal to pick it up. Hellfire also uses value `2` for quest rewards so they can be picked up while they're falling (though actually hitting the pickup in the ten frame animation is a bit difficult).

![2024-07-28T125140_devilutionx](https://github.com/user-attachments/assets/daf69e1f-c44b-4d92-93fb-317d4cdb679a)
(In this screenshot I'm hovering over the left edge of the tile above and to the right of the rock, which highlights it as part of the first two blocks in `TrySelectItem()`)

**Reply by kphoenix137** [reply]
> `hasAnimationFlag` > `isAnimated`, this is used for items that have just been dropped (and are playing the flip animation) or the magic rock, which has a repeating animation. `request` > `isBeingPickedUp`, this is a flag set when a player claims an item and attempts to pick it up
> 
> `selectionFlag` > I don't have a good name for this but this is a combination of two properties about how players can interact with an object. The values are `0` = cannot be interacted with (usually dropped items in the middle of the drop animation), `1` = can be interacted with at ground level, `2` = can be interacted with at elevation, `3` = `1 | 2` = can be interacted with at both ground level and elevation (doesn't appear to actually be used).
> 
> Value `2` is used for the magic rock while it's using the second half of the sprite sheet, which renders the item at a height that makes it look like it's sitting on the pedestal. As this doesn't use the ground level flag the player can't pick the item up by hovering over the tile the object is actually sitting on, instead the cursor highlight logic tests the tile below the cursor making it so the player needs to hover over the tile above the rock/pedestal to pick it up. Hellfire also uses value `2` for quest rewards so they can be picked up while they're falling (though actually hitting the pickup in the ten frame animation is a bit difficult).
> 
> ![2024-07-28T125140_devilutionx](https://private-user-images.githubusercontent.com/357684/352786885-daf69e1f-c44b-4d92-93fb-317d4cdb679a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyMDQ4NTMsIm5iZiI6MTcyMjIwNDU1MywicGF0aCI6Ii8zNTc2ODQvMzUyNzg2ODg1LWRhZjY5ZTFmLWM0NGItNGQ5Mi05M2ZiLTMxN2Q0Y2RiNjc5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyOFQyMjA5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MDY3MmJiZmQ5MDU3MjdiY2NkZjhjNjIwYzRiODIzZTBiOWYwNTAwZDk3ZTE1ZjYxYThlNmQzYTM4M2IwOTBjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6iwbiKbsFXjBNSQuPl75pL8qn04HqyOl3FLSIRHRaLo) (In this screenshot I'm hovering over the left edge of the tile above and to the right of the rock, which highlights it as part of the first two blocks in `TrySelectItem()`)

Updated per your suggestions

**Reply by AJenbo** [reply]
Took care of _iSelFlag since it's part of a wider system related to cursor code, objects and monsters: https://github.com/diasurgical/devilutionX/pull/7416

**Comment by AJenbo** [comment]
List looks good now feel free to apply this.

**Comment by mewmew** [comment]
Here are the names used for the item struct fields in Djavul. Feel free to use any for inspiration : )

Of course, I know DevilX doesn't use abbreviations as much and such, so the style of naming may differ a bit.

```go
// _iAnimFlag
DropAnimPlaying bool
// _iSeed
Seed int32
// _iCreateInfo
ItemCreationFlags d1enum.ItemCreationFlags
// _itype
ItemType d1enum.ItemType
// _ix, _iy
Coordinate
// _iDelFlag
Delete bool // unused
// _iSelFlag
ItemSelect d1enum.ItemSelect
// _iPostDraw
DrawPost bool
// _iIdentified
Identified bool
// _iMagical
ItemQuality d1enum.ItemQuality
// _iName
NormalName string
// _iIName
MagicName string
// _iLoc
ItemLoc d1enum.ItemLoc
// _iClass
ItemClass d1enum.ItemClass
// _iCurs
ItemGraphicID d1enum.ItemGraphicID
// _ivalue
ItemBaseValue int
// _iIvalue
ItemValue int
// _iMinDam, _iMaxDam
MinDamage, MaxDamage int
// _iAC
ArmorClass int
// _iFlags
ItemAbilityFlags d1enum.ItemAbilityFlags
// _iMiscId
ItemMiscID d1enum.ItemMiscID
// _iSpell
SpellID d1enum.SpellID
// _iCharges, _iMaxCharges
CurCharges, MaxCharges int
// _iDurability, _iMaxDur
CurDur, MaxDur int
// _iPLDam
Damage int
// _iPLToHit
ChanceToHitBonus int
// _iPLAC
EnhancedArmorClass int
// _iPLStr
Str int
// _iPLMag
Mag int
// _iPLDex
Dex int
// _iPLVit
Vit int
// _iPLFR
FireResist int
// _iPLLR
LightningResist int
// _iPLMR
MagicResist int
// _iPLMana
MP int
// _iPLHP
HP int
// _iPLDamMod
DamageModifier int
// _iPLGetHit
DamageFromEnemies int
// _iPLLight
LightRadius int
// _iSplLvlAdd
SpellLvlBonus int
// _iRequest
HeldInHand bool
// _iUid
UniqueItemID d1enum.UniqueItemID
// _iFMinDam, _iFMaxDam
MinFireDamage, MaxFireDamage int
// _iLMinDam, _iLMaxDam
MinLightningDamage, MaxLightningDamage int
// _iPLEnAc
EnhancedAccuracy int
// _iPrePower
PrefixItemPowerID d1enum.ItemPowerID
// _iSufPower
SuffixItemPowerID d1enum.ItemPowerID
// _iVAdd1, _iVMult1
PrefixValue, PrefixValueMult int
// _iVAdd2, _iVMult2
SuffixValue, SuffixValueMult int
// _iMinStr, _iMinMag, _iMinDex
StrReq, MagReq, DexReq, VitReq int
// _iStatFlag
Equipable bool
// IDidx
ItemID d1enum.ItemID
```
